tests/unit/models/Component.spec.js

```diff
@@ -539,5 +567,787 @@ describe('Test class: Component', () => {
+          name: 'attribute-string',
+          type: 'String',
+          rules: {
+            min: 5,
+            max: 10,
+            regex: /^[a-z]+$/,
+          },
+        });
+
+        it('Should return false if the "values" rule includes the attribute\'s value', () => {
+          const attributeDefinitionValue = new ComponentAttributeDefinition({
+            name: 'attribute-string',
+            type: 'String',
+            rules: { values: ['test'] },
+          });
+
+          const attribute = new ComponentAttribute({
+            name: 'attribute-string',
+            value: 'test',
+            type: 'String',
+            definition: attributeDefinitionValue,
+          });
+
+          expect(new Component({
+            attributes: [attribute],
+            definition: { definedAttributes: [attributeDefinitionValue] },
+          }).hasError()).toEqual(false);
+        });
+
+        it(
+          'Should return true if the "values" rule doesn\'t includes the attribute\'s value',
+          () => {
+            const attributeDefinitionValue = new ComponentAttributeDefinition({
+              name: 'attribute-string',
+              type: 'String',
+              rules: {
+                values: ['test'],
+              },
+            });
+
+            const attribute = new ComponentAttribute({
+              name: 'attribute-string',
+              value: 'string',
+              type: 'String',
+              definition: attributeDefinitionValue,
+            });
+
+            expect(new Component({
+              attributes: [attribute],
+              definition: {
+                definedAttributes: [attributeDefinitionValue],
+              },
+            }).hasError()).toEqual(true);
+          },
+        );
+
+        it('Should return true if attribute\'s value is smaller than the "min" rule', () => {
+          const attribute = new ComponentAttribute({
+            name: 'attribute-string',
+            value: 'str',
+            type: 'String',
+            definition: attributeDefinition,
+          });
+
+          expect(new Component({
+            attributes: [attribute],
+            definition: {
+              definedAttributes: [attributeDefinition],
+            },
+          }).hasError()).toEqual(true);
+        });
+
+        it('Should return true if attribute\'s value is bigger than the "max" rule', () => {
+          const attribute = new ComponentAttribute({
+            name: 'attribute-string',
+            value: 'stringifies',
+            type: 'String',
+            definition: attributeDefinition,
+          });
+
+          expect(new Component({
+            attributes: [attribute],
+            definition: {
+              definedAttributes: [attributeDefinition],
+            },
+          }).hasError()).toEqual(true);
+        });
+
+        it(
+          'Should return true if the attribute\'s value doesn\'t match with the "regex" rule ',
```

Comment:
doesn\'t match the "regex" rule'

(don't forget to delete the space at the end)

Reply:
forgot to delete 'with'